RockLib.Messaging.Kafka/KafkaReceiver.cs

```diff
@@ -220,34 +221,38 @@ protected override void Dispose(bool disposing)
                 }
             }
 
+            //wait for the polling task to indicate it is okay to dispose of the remaining resources
+            //otherwise, memory access exceptions can occur when the consumer accesses a disposed  cancellation token
+            _completeDispose.Wait();
+
+            _consumerCancellation.Dispose();
+            _trackingCollection.Dispose();
+
             if (_consumer.IsValueCreated)
             {
                 _consumer.Value.Close();
-                _consumer.Value.Dispose();
```

Comment:
Is there an issue with disposing the consumer?

Reply:
Probably not, but technically redundant: https://github.com/confluentinc/confluent-kafka-dotnet/blob/master/src/Confluent.Kafka/Consumer.cs#L558

Reply:
Docs say to prefer `Close` over `Dispose`: https://github.com/confluentinc/confluent-kafka-dotnet/blob/master/src/Confluent.Kafka/IConsumer.cs#L654